test/com/google/enterprise/adaptor/database/TupleReaderTest.java

```diff
@@ -118,6 +118,24 @@ public void testVARCHAR() throws Exception {
     assertEquals(golden, result);
   }
 
+  @Test
+  public void testNullVARCHAR() throws Exception {
+    executeUpdate("create table data(id integer, colname varchar)");
```

Comment:
What's your reasoning for wanting the ID columns? Having a test with multiple columns seems good, as does a test with mixed null/non-null values, but it doesn't need to be replicated for each type. I would drop the ID columns in these null tests and add a single test with a few columns of duplicate and different types, with some nulls mixed in.